src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkRuleOrCheckpoint.kt

```diff
@@ -5,4 +5,17 @@ import com.jetbrains.python.psi.PyTypedElement
 
 interface SmkRuleOrCheckpoint : SmkRuleLike<SmkRuleOrCheckpointArgsSection>, PyTypedElement {
     fun collectWildcards(): List<Pair<PsiElement, String>>
```

Comment:
Also wrong impl of `collectWildcards()` in rule and checkpoint, Visitor pattern isn't used correctly there.